src/org/labkey/remoteapi/assay/GetProtocolCommand.java

```diff
@@ -9,7 +9,7 @@
 public class GetProtocolCommand extends Command<ProtocolResponse>
 {
     private String _providerName;
-    private Integer _protocolId;
+    private Long _protocolId;
```

Comment:
I'd make this a long instead of a Long. We need a value for the API to work, so we should prevent users from passing in null

Reply:
How does that square to the case where we get an instance using the string (providerName) constructor?  Also, the @Override getParameters method checks _protocolId for null to determine whether this instance uses protocolId or providerName

Reply:
@labkey-jeckels The case for having _protocolId be null is to get the template for an assay provider. The assay-getProtocol action when call with just a provider name and no protocolId will get you the template for that assay. I believe that is the reason for using "Long" here instead of "long", right?

Reply:
Keeping the member variable Long, updating constructors and setters that access it long

Reply:
Chris and I just talked over IM. I didn't realize it was either/or with the protocolId vs the name. I recommended keeping the member variable as a Long, but changing the constructor argument to a long, to signal if you're opting for the Id approach, you have to supply a value

Reply:
Yes, changing the constructor argument makes sense. Thanks.